### PR TITLE
Ana/161 claim rewards

### LINKED ISSUE #161: claim rewards
- add button back to balances section
- claim rewards action modal

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #161 

Debugging this:

image

### CODE REVIEW

**Comment by Bitcoinera** [comment]
Ouch, wanted to create it as a draft. Clicked too fast...

**Comment by Bitcoinera** [comment]
OK, let's test this. Actually I can test this on simply on Cosmos Hub :shrug:

**Comment by Bitcoinera** [comment]
Tested it and it is working

File: components/ActionModals/ClaimModal.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      }
+    },
+    validatorsWithRewards() {
+      if (this.rewards) {
+        return this.rewards.length > 0
+      } else {
+        return false
+      }
+    },
+    feeDenom() {
```
Comment (faboweb) [inline components/ActionModals/ClaimModal.vue:108]: can we simplify this now?

File: network.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+require('dotenv').config()
```
Comment (faboweb) [inline network.js:1]: why did you add this?

File: nuxt.config.js [hunk 3]
```diff
@@ -69,5 +71,11 @@ export default {
   },
 
   // Build Configuration (https://go.nuxtjs.dev/config-build)
-  build: {},
+  build: {
```
Comment (faboweb) [inline nuxt.config.js:74]: ?

File: signing/transaction-manager.js [hunk 4]
```diff
@@ -138,3 +138,27 @@ export async function pollTxInclusion(txHash, iteration = 0) {
     )
   }
 }
+
+// limitation of the Ledger Nano S, so we pick the top 5 rewards and inform the user.
```
Comment (faboweb) [inline signing/transaction-manager.js:142]: can we put this in another file? doesn'T really belong in the transaction manager

File: components/ActionModals/ClaimModal.vue [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    :selected-denom="feeDenom"
+    title="Claim Rewards"
+    class="modal-withdraw-rewards"
+    submission-error-prefix="Withdrawal failed"
+    feature-flag="claim_rewards"
+    :transaction-type="lunieMessageTypes.CLAIM_REWARDS"
+    :rewards="rewards"
+    :disable="validatorsWithRewards"
+  >
+    <span class="form-message withdraw-limit">
```
Comment (faboweb) [inline components/ActionModals/ClaimModal.vue:16]: let's do this only if the user is logged in with Ledger

Comment (Bitcoinera) [reply components/ActionModals/ClaimModal.vue]: we also don't need to restrict to `top5Validators` if it is not logged in with Ledger

File: signing/transaction-manager.js (hunk 4 above)
Comment (Bitcoinera) [reply signing/transaction-manager.js]: Done

File: network.js (hunk 2 above)
Comment (Bitcoinera) [reply network.js:1]: this is from a previous state, trying to use Gaia Testnet from the Figment's datahub (it needs an API key)

File: components/ActionModals/ClaimModal.vue (hunk 5 above)
Comment (faboweb) [reply components/ActionModals/ClaimModal.vue]: That is what I meant

File: components/ActionModals/ClaimModal.vue (hunk 1 above)
Comment (Bitcoinera) [reply components/ActionModals/ClaimModal.vue:121]: well, do you know if in Stargate you can pay fees with IBC tokens? Probably, no? Like in Terra and e-Money

Comment (faboweb) [reply components/ActionModals/ClaimModal.vue:121]: Fee denoms get selected in the table invoice as far as I know

Comment (Bitcoinera) [reply components/ActionModals/ClaimModal.vue:121]: actually I'm thinking, this shouldn't be here. It should simply default to stakingDenom :thinking: Since this is the only currency you can stake with & logically claim

Comment (Bitcoinera) [reply components/ActionModals/ClaimModal.vue:121]: we have that logic only for SendModal. Let me check

**Comment by Bitcoinera** [comment]
Ooops

image

**Comment by Bitcoinera** [comment]
It's working. Only styles need to be fixed (comment above)

File: common/ledger.js [hunk 6]
```diff
@@ -20,8 +20,8 @@ export async function getLedger() {
   return ledger
 }
 
-// limitation of the Ledger Nano S, so we pick the top 5 rewards and inform the user.
-export function getTop5RewardsValidators(rewards) {
+// limitation of the Ledger Nano S: if top5 is true, we pick the top 5 rewards and inform the user.
+export function getRewardsValidators(rewards, top5) {
```
Comment (faboweb) [inline common/ledger.js:24]: nice
